packages/core/src/locales/pt.ts

```diff
@@ -0,0 +1,123 @@
+import type { $ZodStringFormats } from "../checks.js";
+import type * as errors from "../errors.js";
+import * as util from "../util.js";
+
+const Sizable: Record<string, { unit: string; verb: string }> = {
+  string: { unit: "caracteres", verb: "ter" },
+  file: { unit: "bytes", verb: "ter" },
+  array: { unit: "elementos", verb: "ter" },
+  set: { unit: "elementos", verb: "ter" },
+};
+
+function getSizing(origin: string): { unit: string; verb: string } | null {
+  return Sizable[origin] ?? null;
+}
+
+export const parsedType = (data: any): string => {
+  const t = typeof data;
+
+  switch (t) {
+    case "number": {
+      return Number.isNaN(data) ? "NaN" : "número";
+    }
+    case "object": {
+      if (Array.isArray(data)) {
+        return "array";
+      }
+      if (data === null) {
+        return "nulo";
+      }
+      if (Object.getPrototypeOf(data) !== Object.prototype) {
```

Comment:
Make the same check of the original: https://github.com/colinhacks/zod/blob/v4/packages/core/src/locales/en.ts#L31C11-L31C79
```suggestion
      if (Object.getPrototypeOf(data) !== Object.prototype && data.constructor) {
```